keyboards/splitkb/kyria/keymaps/artflag/keymap.c

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 Your Name (@artflag)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+// Tap Dance declarations
+
+enum {
+    CT_CLN, // Reverses colon and semicolon and triggers semicolon on tap dance.
+};
+
+void dance_cln_finished(qk_tap_dance_state_t *state, void *user_data) {
+    if (state->count == 1) {
+        register_code16(KC_COLN);
+    } else {
+        register_code(KC_SCLN);
+    }
+}
+
+void dance_cln_reset(qk_tap_dance_state_t *state, void *user_data) {
+    if (state->count == 1) {
+        unregister_code16(KC_COLN);
+    } else {
+        unregister_code(KC_SCLN);
+    }
+}
+
+// All tap dance functions would go here.
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [CT_CLN] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_cln_finished, dance_cln_reset),
```

Comment:
Just a heads up, you don't need the advanced function here. `ACTION_TAP_DANCE_DOUBLE(KC_SCLN, KC_COLN)` should work here.

Reply:
it doesnt seem to work, also this snippet is straight from the docs, so I can send a docs update at the same time if the current solution is not the best one.

Reply:
no worries.